File: SretneSapice/SretneSapice.Services/Database/ServiceRequest.cs
using System;
using System.Collections.Generic;

namespace SretneSapice.Services.Database;

public partial class ServiceRequest
{
    public int ServiceRequestId { get; set; }

    public int? DogWalkerId { get; set; }

    public int? UserId { get; set; }

    public DateTime StartTime { get; set; }

    public DateTime EndTime { get; set; }

    public DateTime? Date { get; set; }

    public string? Status { get; set; }
    public string DogBreed { get; set; }
    public bool LiveLocationEnabled { get; set; }

    public virtual DogWalker? DogWalker { get; set; }

    public virtual User? User { get; set; }
}

File: SretneSapice/UI/sretnesapice_mobile/lib/models/service_request.dart
import 'package:json_annotation/json_annotation.dart';
import 'package:sretnesapice_mobile/models/dog_walker.dart';
import 'package:sretnesapice_mobile/models/user.dart';

part 'service_request.g.dart';

@JsonSerializable()
class ServiceRequest {
  int? serviceRequestId;
  int? dogWalkerId;
  int? userId;
  DateTime? startTime;
  DateTime? endTime;
  DateTime? date;
  String? status;
  String? dogBreed;
  bool? liveLocationEnabled;
  User? user;
  DogWalker? dogWalker;

  ServiceRequest(
      this.serviceRequestId,
      this.dogWalkerId,
      this.userId,
      this.startTime,
      this.endTime,
      this.date,
      this.status,
      this.dogBreed,
      this.liveLocationEnabled,
      this.user,
      this.dogWalker);

  factory ServiceRequest.fromJson(Map<String, dynamic> json) =>
      _$ServiceRequestFromJson(json);

  /// `toJson` is the convention for a class to declare support for serialization
  /// to JSON. The implementation simply calls the private, generated
  /// helper method `_$UserToJson`.
  Map<String, dynamic> toJson() => _$ServiceRequestToJson(this);
}
